digital_asset_types/src/dapi/common/asset.rs

```diff
@@ -348,36 +349,32 @@ pub fn asset_to_rpc(asset: FullAsset, options: &Options) -> Result<RpcAsset, DbE
         authorities,
         creators,
         groups,
+        token_info,
     } = asset;
     let rpc_authorities = to_authority(authorities);
     let rpc_creators = to_creators(creators);
     let rpc_groups = to_grouping(groups, options)?;
     let interface = get_interface(&asset)?;
+    let content = get_content(&data);
+    let mut chain_data_selector_fn = jsonpath_lib::selector(&data.chain_data);
+    let chain_data_selector = &mut chain_data_selector_fn;
 
-    let (content, edition_nonce, basis_points, mutable, uses) = if let Some(data) = &data {
-        let mut chain_data_selector_fn = jsonpath_lib::selector(&data.chain_data);
-        let chain_data_selector = &mut chain_data_selector_fn;
-        (
-            get_content(data),
-            safe_select(chain_data_selector, "$.edition_nonce").and_then(|v| v.as_u64()),
-            safe_select(chain_data_selector, "$.primary_sale_happened")
-                .and_then(|v| v.as_bool())
-                .unwrap_or(false),
-            data.chain_data_mutability.clone().into(),
-            data.chain_data.get("uses").map(|u| Uses {
-                use_method: u
-                    .get("use_method")
-                    .and_then(|s| s.as_str())
-                    .unwrap_or("Single")
-                    .to_string()
-                    .into(),
-                total: u.get("total").and_then(|t| t.as_u64()).unwrap_or(0),
-                remaining: u.get("remaining").and_then(|t| t.as_u64()).unwrap_or(0),
-            }),
-        )
-    } else {
-        (None, None, false, false, None)
-    };
+    let edition_nonce =
+        safe_select(chain_data_selector, "$.edition_nonce").and_then(|v| v.as_u64());
+    let basis_points = safe_select(chain_data_selector, "$.primary_sale_happened")
+        .and_then(|v| v.as_bool())
+        .unwrap_or(false);
+    let mutable = data.chain_data_mutability.clone().into();
+    let uses = data.chain_data.get("uses").map(|u| Uses {
+        use_method: u
+            .get("use_method")
+            .and_then(|s| s.as_str())
+            .unwrap_or("Single")
+            .to_string()
+            .into(),
+        total: u.get("total").and_then(|t| t.as_u64()).unwrap_or(0),
+        remaining: u.get("remaining").and_then(|t| t.as_u64()).unwrap_or(0),
+    });
```

Comment:
In #214 data was changed to options because fungible assets may not have associated metadata but after reviewing variations all on-chain token standards TE, core, TokenK + Metdata provide methods for populating the `asset_data` thus DAS will now return assets that are "fungible" but only if they have metadata information.

Reply:
That seems fine with me.  Some don't have token standards is the main thing I think of with old fungible assets